src/wallet/dump.cpp

```diff
@@ -78,3 +78,167 @@ bool DumpWallet(std::shared_ptr<CWallet> wallet, bilingual_str& error)
+
+    // dummy chain interface
+    std::shared_ptr<CWallet> wallet(new CWallet(nullptr /* chain */, name, std::move(database)), WalletToolReleaseWallet);
+    LOCK(wallet->cs_wallet);
+    bool first_run = true;
+    DBErrors load_wallet_ret = wallet->LoadWallet(first_run);
+    if (load_wallet_ret != DBErrors::LOAD_OK) {
+        error = strprintf(_("Error creating %s"), name);
+        return false;
+    }
+
+    // Get the database handle
+    WalletDatabase& db = wallet->GetDatabase();
+    std::unique_ptr<DatabaseBatch> batch = db.MakeBatch();
+    batch->TxnBegin();
+
+    // Compute the checksum
+    CHashWriter hasher(0, 0);
+    uint256 checksum;
+
+    // Read the records from the dump file and write them to the database
+    bool ret = true;
+    bool first = true;
+    while (dump_file.good()) {
+        std::string key;
+        std::getline(dump_file, key, ',');
+        std::string value;
+        std::getline(dump_file, value, '\n');
+
+        if (key == "checksum") {
+            std::vector<unsigned char> parsed_checksum = ParseHex(value);
+            std::copy(parsed_checksum.begin(), parsed_checksum.end(), checksum.begin());
+            break;
+        }
+
+        std::string line = strprintf("%s,%s\n", key, value);
+        hasher.write(line.data(), line.size());
+
+        if (key.empty() || value.empty()) {
+            continue;
+        }
+
+        // Check magic for first record
+        if (first) {
+            if (key != DUMP_MAGIC) {
+                error = strprintf(_("Error: Dumpfile identifier record is incorrect. Got \"%s\", expected \"%s\"."), key, DUMP_MAGIC);
+                ret = false;
+                break;
+            }
+            // Check the version number (value of first record)
+            uint32_t ver;
+            if (!ParseUInt32(value, &ver)) {
+                error =strprintf(_("Error: Unable to parse version %u as a uint32_t"), value);
+                ret = false;
+                break;
+            }
+            if (ver != DUMP_VERSION) {
+                error = strprintf(_("Error: Dumpfile version is not supported. This version of bitcoin-wallet only supports version 1 dumpfiles. Got dumpfile with version %s"), value);
+                ret = false;
+                break;
+            }
+            first = false;
+            continue;
+        }
+
+        if (!IsHex(key)) {
+            error = strprintf(_("Error: Got key that was not hex: %s"), key);
+            ret = false;
+            break;
+        }
+        if (!IsHex(value)) {
+            error = strprintf(_("Error: Got value that was not hex: %s"), value);
+            ret = false;
+            break;
+        }
+
+        std::vector<unsigned char> k = ParseHex(key);
+        std::vector<unsigned char> v = ParseHex(value);
+
+        CDataStream ss_key(k, SER_DISK, CLIENT_VERSION);
+        CDataStream ss_value(v, SER_DISK, CLIENT_VERSION);
+
+        batch->Write(ss_key, ss_value);
```

Comment:
In commit "wallettool: Add createfromdump command" (6137750082dd28bf38bbb2922d6d040d4e854643)

`ret` should be set to false if the Write call fails

Reply:
Done